sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/SyncRestProxy.java

```diff
@@ -161,10 +161,9 @@ private Object handleBodyReturnType(final HttpResponseDecoder.HttpDecodedRespons
 
         final Object result;
         if (httpMethod == HttpMethod.HEAD
-            && (TypeUtil.isTypeOrSubTypeOf(
-            entityType, Boolean.TYPE) || TypeUtil.isTypeOrSubTypeOf(entityType, Boolean.class))) {
-            boolean isSuccess = (responseStatusCode / 100) == 2;
-            result = isSuccess;
+            && (TypeUtil.isTypeOrSubTypeOf(entityType, Boolean.TYPE)
+            || TypeUtil.isTypeOrSubTypeOf(entityType, Boolean.class))) {
+            result = (responseStatusCode - 200) < 100; // No more division!
```

Comment:
This might not work as expected https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#information_responses